gatsby-node.js

```diff
@@ -333,24 +331,27 @@ async function getIndexToUse({ index, tempIndex, enablePartialUpdates }) {
 }
 
 async function getSettingsToApply({
-  settings: givenSettings,
+  settings,
   index,
   tempIndex,
   indexToUse,
 }) {
-  const { replicaUpdateMode, ...settings } = givenSettings;
   const existingSettings = await index.getSettings().catch(e => {
     report.panic(`${e.toString()} ${index.indexName}`);
   });
 
+  if (!settings) {
+    return existingSettings;
+  }
+
   const replicasToSet = getReplicasToSet(
     settings.replicas,
     existingSettings.replicas,
-    replicaUpdateMode
+    settings.replicaUpdateMode
   );
 
   const requestedSettings = {
-    ...(settings ? settings : existingSettings),
+    ...settings,
```

Comment:
replicaUpdateMode needs to be removed from this object, since it's not a valid setting. Really this is a problem and replicaUpdateMode should have been an option separate from settings

Reply:
Gotcha, I can add a check for that. Feels like we should have had `indexSettings` and `options` or similar

Reply:
I just tested this in my local `node_modules` folder replacing `gatsby-node.js` and it seemed to work. I remove the `settings` in my `queries` array